src/vulkan/resource.h

```diff
@@ -31,10 +31,14 @@ class Resource {
     return host_accessible_memory_;
   }
 
+  virtual void CopyToHost(VkCommandBuffer command) = 0;
+
   virtual void Shutdown();
 
   void* HostAccessibleMemoryPtr() const { return memory_ptr_; }
 
+  size_t GetSize() const { return size_; }
```

Comment:
What are the units?  bytes?
What are you measuring? the whole underlying storage ?